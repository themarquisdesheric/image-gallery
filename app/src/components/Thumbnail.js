import React, { Component } from 'react';
import PropTypes from 'prop-types';
import DeleteButton from './DeleteButton';

class Thumbnail extends Component {
  static propTypes = { 
    data: PropTypes.array.isRequired,
    onDelete: PropTypes.func.isRequired
  }

  render() {
    const { data, onDelete } = this.props;

    return (
      <ul>
        {data.map((img, i) => (
          <li key={i} 
              className="Thumbnail"
          >
            <div>
              <img src={img.url} 
                   alt={img.title} 
                   className="Thumbnail-image"
              />
            </div>
            <div className="Thumbnail-title">
              {img.title}
            </div>
            <DeleteButton onClick={() => onDelete(img._id)} />
          </li>
        ))}
      </ul>
    );
  }
}

export default Thumbnail;