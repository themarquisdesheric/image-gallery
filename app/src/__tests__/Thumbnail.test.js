import React from 'react';
import renderer from 'react-test-renderer';
import Thumbnail from '../components/Thumbnail';
import images from './_data';

function onDelete() {}

describe('Thumbnail', () => {
  it('renders thumbnails', () => {
    const tree = renderer.create(<Thumbnail data={images} onDelete={onDelete} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});