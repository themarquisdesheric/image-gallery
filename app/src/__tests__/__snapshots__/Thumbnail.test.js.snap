// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Thumbnail renders thumbnails 1`] = `
<ul>
  <li
    className="Thumbnail"
  >
    <div>
      <img
        alt="Black & Tan"
        className="Thumbnail-image"
        src="http://f.cl.ly/items/3g3J1G0w122M360w380O/3726490195_f7cc75d377_o.jpg"
      />
    </div>
    <div
      className="Thumbnail-title"
    >
      Black & Tan
    </div>
    <button
      className="delete-button"
      onClick={[Function]}
    >
      Delete ðŸ—‘
    </button>
  </li>
  <li
    className="Thumbnail"
  >
    <div>
      <img
        alt="Falling Bunny"
        className="Thumbnail-image"
        src="http://static.boredpanda.com/blog/wp-content/uploads/2015/09/cute-bunnies-25__605.jpg"
      />
    </div>
    <div
      className="Thumbnail-title"
    >
      Falling Bunny
    </div>
    <button
      className="delete-button"
      onClick={[Function]}
    >
      Delete ðŸ—‘
    </button>
  </li>
  <li
    className="Thumbnail"
  >
    <div>
      <img
        alt="Marshmellow Puff"
        className="Thumbnail-image"
        src="http://static.boredpanda.com/blog/wp-content/uploads/2015/09/cute-bunnies-110__605.jpg"
      />
    </div>
    <div
      className="Thumbnail-title"
    >
      Marshmellow Puff
    </div>
    <button
      className="delete-button"
      onClick={[Function]}
    >
      Delete ðŸ—‘
    </button>
  </li>
</ul>
`;
